extensions/libarchive/MergeContent.cpp

```diff
@@ -342,6 +362,56 @@ std::shared_ptr<core::FlowFile> ZipMerge::merge(core::ProcessContext *context, c
   return flowFile;
 }
 
+void AttributeMerger::mergeAttributes() {
+  std::map<std::string, std::string> commonAttributes = getCommonAttributes();
+
+  for (const auto& pair : commonAttributes) {
+    session_->putAttribute(mergeFlow_, pair.first, pair.second);
+  }
+}
+
+std::map<std::string, std::string> AttributeMerger::getCommonAttributes() {
+  std::map<std::string, std::string> commonAttributes;
+  bool isFirst = true;
+  for (const auto& flow : flows_) {
+    if (isFirst) {
+      commonAttributes = flow->getAttributes();
+      isFirst = false;
+    } else {
+      processFlowFile(flow, commonAttributes);
+    }
+  }
+  return commonAttributes;
+}
+
+void KeepOnlyCommonAttributesMerger::processFlowFile(const std::shared_ptr<core::FlowFile> &flow, std::map<std::string, std::string>& commonAttributes) {
+  auto flowAttributes = flow->getAttributes();
+  for (auto it = commonAttributes.cbegin(); it != commonAttributes.cend();) {
+    if (flowAttributes.find(it->first) != flowAttributes.end()) {
```

Comment:
with `FlowFile::getAttribute` we could both check for existence and extract the value in one motion (here for `flowAttributes` and in the other `processFlowFile` for `commonAttributes`)